### PR TITLE
Add implements assertion for cache

### PR DESCRIPTION
**Pull request by pmorie**
@lavalamp

### CODE REVIEW

File: pkg/client/cache/store.go [hunk 1]
```diff
@@ -116,6 +116,8 @@ type cache struct {
 	keyFunc KeyFunc
 }
 
+var _ Store = &cache{}
```
Comment (lavalamp) [inline pkg/client/cache/store.go]: I have an extremely minor preference for `var _ = Store(&cache{})`-- I think that's more common in our code base?

**Comment by lavalamp** [comment]
LGTM-- feel free to apply label yourself after fixing nit (or deciding not to)

File: pkg/client/cache/store.go (hunk 1 above)
Comment (pmorie) [inline pkg/client/cache/store.go]: Hrm, I will do a quick, highly unscientific analysis

Comment (pmorie) [inline pkg/client/cache/store.go]: @lavalamp looks like your style is in the minority from this:

```
$ grep -nr 'var _ ' pkg/ | grep '(' | wc -l
36

★ pmorie:~/code/kubernetes (master) 
$ grep -nr 'var _ ' pkg/ | grep -v '(' | wc -l
113
```

Comment (lavalamp) [inline pkg/client/cache/store.go]: hm, that leaves out `var ( ... )` blocks, but it probably doesn't matter if
the difference is that big. Never mind, then.

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Pull request merged**